runtime/parachains/src/paras_inherent.rs

```diff
@@ -627,31 +637,45 @@ impl<T: Config> Pallet<T> {
 				let freed = collect_all_freed_cores::<T, _>(freed_concluded.iter().cloned());
 
 				<scheduler::Pallet<T>>::clear();
-				<scheduler::Pallet<T>>::schedule(freed, <frame_system::Pallet<T>>::block_number());
+				let now = <frame_system::Pallet<T>>::block_number();
+				<scheduler::Pallet<T>>::schedule(freed, now);
 
 				let scheduled = <scheduler::Pallet<T>>::scheduled();
 
+				let relay_parent_number = now - One::one();
+
+				let checker_ctx = CandidateCheckContext::<T>::new(now, relay_parent_number);
```

Comment:
I could definitely see `CandidateCheckContext` being updated for contextual execution, although we might need to instantiate it on a candidate-by-candidate basis.